scripts/extend_public_ids.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Script to update PublicIds to ExtendedPublicIds for package dependencies."""
+
+import json
+from pathlib import Path
+from typing import Dict, List, Set
+
+import click
+
+from aea.configurations.constants import (
+    AGENT,
+    CONNECTION,
+    CONTRACT,
+    DEFAULT_AEA_CONFIG_FILE,
+    DEFAULT_CONNECTION_CONFIG_FILE,
+    DEFAULT_CONTRACT_CONFIG_FILE,
+    DEFAULT_PROTOCOL_CONFIG_FILE,
+    DEFAULT_SKILL_CONFIG_FILE,
+    PROTOCOL,
+    SKILL,
+)
+from aea.configurations.data_types import ExtendedPublicId, PublicId
+from aea.helpers.ipfs.base import IPFSHashOnly
+import yaml
+from aea.helpers.yaml_utils import _AEAYamlDumper, _AEAYamlLoader
+
+PACKAGES_PATH = Path(__file__).parent.parent / "packages"
+COMPONENTS = [
+    (AGENT, DEFAULT_AEA_CONFIG_FILE),
+    (SKILL, DEFAULT_SKILL_CONFIG_FILE),
+    (CONTRACT, DEFAULT_CONTRACT_CONFIG_FILE),
+    (CONNECTION, DEFAULT_CONNECTION_CONFIG_FILE),
+    (PROTOCOL, DEFAULT_PROTOCOL_CONFIG_FILE),
+]
+
+COMPONENT_TO_FILE = dict(COMPONENTS)
+
+
+def to_plural(string: str) -> str:
+    """Convert component to plural"""
+    return string + "s"
+
+
+def reduce_set(list_of_sets: List[Set]) -> Set:
+    """Reduce a list of sets to one dimentional set."""
+    reduced = set()
+    for s in list_of_sets:
+        reduced = reduced.union(s)
+
+    return reduced
+
+
+def to_public_id(public_id: str) -> str:
+    """Convert to public id."""
+    try:
+        return str(PublicId.from_str(public_id))
+    except ValueError:
+        return str(PublicId.from_json(ExtendedPublicId.from_str(public_id).json))
+
+
+def normalize_package(public_id: str, separator: str = "-") -> str:
+    """Convert to public id."""
+
+    component_type, component_id = public_id.split(separator)
+    component_id = to_public_id(component_id)
+    return component_type + separator + component_id
+
+
+def generate_dependency_tree(packages_dir: Path) -> List[List[str]]:
```

Comment:
The decency resolver from aea is not much of use when working with whole registry so I implemented this function to create a dependency tree of every available package in registry

Reply:
Ok. Then I would move this utility into AEA nevertheless and import here. IT will be useful in many contexts